virtcontainers/network.go

```diff
@@ -413,6 +469,8 @@ type NetworkNamespace struct {
 	NetNsPath    string
 	NetNsCreated bool
 	Endpoints    []Endpoint
+	Interfaces   []grpc.Interface
+	Routes       []*grpc.Route
```

Comment:
Why do you need to add those fields ? `NetworkNamespace` structure already have the list of `Endpoints`, and each Endpoint has a getter function called `Properties()` returning the `NetworkInfo` related to the sandbox.

Reply:
I see Endpoints store most of these info. But not all.
1. Need to refactor https://github.com/kata-containers/runtime/blob/ff7b4f6ed7f622217f456a9215f5e7c6215ff916/virtcontainers/network.go#L1136 to pass grpc.Interface.Device as tapName. This is easy. 
2. Endpoint includes netlink.Route which does not have a Device field. So we will lost grpc.Route.Device info. Though this field is unused exactly now (just used in agent logs).

Reply:
I agree with @sboeuf , I think it's weird to introduce a new `grpc.Interface` , and  I guess it's better to rework a little of `Endpoint` to involve all the necessary infos